pkg/products/marin3r/reconciler.go

```diff
@@ -129,6 +151,74 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		return phase, nil
 	}
 
+	logrus.Infof("about to start reconciling the discovery service")
+	phase, err = r.reconcileDiscoveryService(ctx, client, productNamespace, installation.Spec.NamespacePrefix)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile DiscoveryService cr"), err)
+		return phase, err
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) reconcileRedis(ctx context.Context, client k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+	logrus.Info("Creating backend redis instance")
+	ns := r.Config.GetNamespace()
+	redisName := fmt.Sprintf("%s%s", constants.RateLimitRedisPrefix, r.installation.Name)
+	rateLimitRedis, err := croUtil.ReconcileRedis(ctx, client, defaultInstallationNamespace, r.installation.Spec.Type, croUtil.TierProduction, redisName, ns, redisName, ns, func(cr metav1.Object) error {
+		owner.AddIntegreatlyOwnerAnnotations(cr, r.installation)
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to reconcile backend redis request: %w", err)
+	}
+
+	r.redisSecretName = rateLimitRedis.Spec.SecretRef.Name
```

Comment:
The way the RateLimitReconciler is implemented expects the Redis secret to have a URL field [\[1\]](https://github.com/integr8ly/integreatly-operator/pull/1215/commits/e8d6598690a4ccba24a70d03f598e4d226eb066b#diff-898ea6af7f0284619a8a48f611898689R183) with the host and port of the redis instance. This is done following the existing implementation for other Redis resources [\[2\]](https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/threescale/reconciler.go#L796-L819), where the operator retrieves the secret created from the CRO and creates a new secret with `URL` concatenating the `uri` and `host` fields from the secret created by CRO.
There's two possible approaches to make it work:
1. To create a new secret as it's done in the 3scale reconciler, setting the `URL` field
2. To modify the RateLimitReconciler to use the `uri` and `port` fields instead of the `URL` 

I don't see a reason not to go for option 2 and save us from copying the secret, maybe someone who knows better than me can tell

Reply:
Thanks @sergioifg94 I'm closing this PR and opening https://github.com/integr8ly/integreatly-operator/pull/1220 I've taken the 1. approach as there was a precedent in the code but I can change to 2. if you think there is a need?

Reply:
Yeah, I think it's safer to go with 1. I don't know exactly what's the reasoning behind but if it was done that way I'd assume it's for a good reason. I'll have a look at the new PR